modules/data.atmosphere/R/check_met_input.R

```diff
@@ -0,0 +1,97 @@
+#' Check a meteorology data file for compliance with the PEcAn standard
+#'
+#' @param metfile Path of met file to check, as a scalar character.
+#' @param variable_table `data.frame` linking standard names to their
+#'   units. Defaults to [pecan_standard_met_table].
+#' @return `TRUE` (invisibly) if successful, otherwise throw an informative error.
+#' @author Alexey Shiklomanov
+#' @export
+check_met_input_file <- function(metfile, variable_table = pecan_standard_met_table) {
+
+  metfile <- normalizePath(metfile)
+
+  PEcAn.logger::severeifnot(
+    file.exists(metfile),
+    msg = glue::glue("File '{metfile}' does not exist.")
+  )
+
+  nc <- ncdf4::nc_open(metfile)
+
+  test_dims <- tryCatch(
+    testthat::test_that(
+      "Dimensions are correct.",
+      {
+        dimensions <- nc[["dim"]]
+        testthat::expect_type(dimensions, "list")
+        testthat::expect_equal(length(dimensions), 3)
+        testthat::expect_true("time" %in% names(dimensions))
+        testthat::expect_match(
+          ncdf4::ncatt_get(nc, "time", "units")[["value"]],
+          "^days since [[:digit:]]{4}-[[:digit:]]{2}-[[:digit:]]{2}T[[:digit:]]{2}:[[:digit:]]{2}:[[:digit:]]{2}Z$"
+        )
+         testthat::expect_true("latitude" %in% names(dimensions))
+         testthat::expect_equal(
+           ncdf4::ncatt_get(nc, "latitude", "units")[["value"]],
+           "degrees_north"
+         )
+         testthat::expect_true("longitude" %in% names(dimensions))
+         testthat::expect_equal(
+           ncdf4::ncatt_get(nc, "longitude", "units")[["value"]],
+           "degrees_east"
+         )
+       }
+    ), error = function(e) conditionMessage(e)
+  )
+
+  test_vars <- purrr::map(
+    names(nc$var),
+    purrr::safely(check_met_variable),
+    nc = nc,
+    variable_table = variable_table
+  ) %>%
+    setNames(names(nc$var)) %>%
+    purrr::transpose()
+
+  test_vars_which_errors <- !purrr::map_lgl(test_vars[["error"]], is.null)
+  test_vars_errors <- test_vars[["error"]][test_vars_which_errors] %>%
+    purrr::map(conditionCall)
+  all_errors <- c(list(dimensions = test_dims), test_vars_errors)
+
+  if (length(all_errors) > 0) {
+    cat(unlist(all_errors), sep = "\n\n")
+    stop("At least one error was found.")
```

Comment:
If we need PEcAn.logger::severeifnot at the top of the function, shouldn't we use it here too? Or conversely, if stop() is OK here can we get away with stopifnot() up top?

Reply:
Ah, good catch. I should have used `PEcAn.logger::logger.severe` here. I'll change that accordingly.

Reply:
I don't want to use the shorthand `severeifnot` because I want to neatly print out all the errors first.